File: C#/Two Pointer/283. Move Zeroes.cs
/*  283. Move Zeroes
Difficulty: Easy
Submission Link: https://leetcode.com/problems/move-zeroes/submissions/1246678906 */

public class Solution {
    public void MoveZeroes(int[] nums) {

        int left = 0;

        for(int i=0; i<nums.Length; i++)
        {
            if (nums[i] != 0)
            {
                (nums[left],nums[i]) = (nums[i], nums[left]);
                left++;
            }
        }
    }
}

File: Python/Two Pointer/283. Move Zeroes.py
"""  283. Move Zeroes
Difficulty: Easy
Submission Link: https://leetcode.com/problems/move-zeroes/submissions/1246678513/ """

class Solution(object):
    def moveZeroes(self, nums):
        zeroIndex = 0
        currIndex = 0
        for i,n in enumerate(nums):
            if nums[zeroIndex] != 0 and n == 0:
                zeroIndex = i
            elif n != 0:
                nums[zeroIndex], nums[i] = nums[i], nums[zeroIndex]
                zeroIndex += 1          


        
        